packages/viewer/src/lib/rendering/canvas.ts

```diff
@@ -121,6 +129,38 @@ export function createCanvasRenderer(): CanvasRenderer {
   };
 }
 
+export function createHiddenCanvasRenderer(): CanvasRenderer {
+  const hiddenCanvas = document.createElement('canvas');
+  let lastFrameNumber: number | undefined;
+
+  return async (data) => {
+    const frameNumber = data.frame.sequenceNumber;
+    const frameIsNewer =
+      lastFrameNumber == null || frameNumber > lastFrameNumber;
+    const frameDimensionsMatch =
+      data.dimensions == null ||
+      Dimensions.isEqual(
+        data.dimensions,
+        data.frame.image.imageAttr.frameDimensions
+      );
+    const image = await loadImageBytes(data.frame.image.imageBytes);
+
+    hiddenCanvas.width = data.canvasDimensions.width;
+    hiddenCanvas.height = data.canvasDimensions.height;
+    const ctx = hiddenCanvas.getContext('2d');
+
+    if (ctx != null && frameIsNewer && frameDimensionsMatch) {
+      lastFrameNumber = frameNumber;
+      drawImage(image, { ...data, canvas: ctx });
+      data.beforeDraw?.();
+      data.canvas.drawImage(hiddenCanvas, 0, 0);
+    }
+
+    image.dispose();
+    return data.frame;
+  };
+}
+
```

Comment:
I think part of the jumping has to do with us using a flex layout on the canvas container. This is also causing an issue where the viewer wouldn't resize to a smaller dimension when placed in a flex container. Anyways, seems like this hidden canvas isn't needed (only did some quick testing) when I replace the CSS for `.canvas-container` with: 

```css
.canvas-container {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--image-background, var(--viewer-background, #ffffff));
}
```